src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,180 @@
+import classNames from 'classnames';
+import {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import * as todoService from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../TodosContext';
+import { ErrorContext } from '../../ErrorContext';
+import { Error } from '../../types/ErrorMessage';
+
+const ERROR_DELAY = 3000;
+
+type Props = {
+  todo: Todo;
+  loadingIds: number[];
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loadingIds,
+}) => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
it's a bit not consistent for me because you have `loadingIds` prop
but not critical I guess in this case